### PR TITLE
KAFKA-16548: Change to use `streamingIterator` 

### PR DESCRIPTION
**Pull request by m1a2st**
change to use `streamingIterator` because we don't show all record, and we need early return

### CODE REVIEW

**Comment by m1a2st** [comment]
@chia7712 please review thanks

File: core/src/main/java/kafka/log/remote/RemoteLogManager.java [hunk 1]
```diff
@@ -463,7 +464,9 @@ private Optional<FileRecords.TimestampAndOffset> lookupTimestamp(RemoteLogSegmen
                 RecordBatch batch = remoteLogInputStream.nextBatch();
                 if (batch == null) break;
                 if (batch.maxTimestamp() >= timestamp && batch.lastOffset() >= startingOffset) {
-                    for (Record record : batch) {
+                    Iterator<Record> recordStreamingIterator = batch.streamingIterator(BufferSupplier.NO_CACHING);
```
Comment (chia7712) [inline core/src/main/java/kafka/log/remote/RemoteLogManager.java]: Could you please close the `recordStreamingIterator` when returning.

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for your contribution. one major comment left.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@m1a2st thanks for your contribution!

**Pull request merged**